### PR TITLE
Generate absolute paths

### PR DESCRIPTION
**Pull request by me-kell**
See https://github.com/plone/cookiecutter-zope-instance/issues/5#issue-1446551887

### CODE REVIEW

**Review by jensens — COMMENTED** [review]
-1 here, see https://github.com/plone/cookiecutter-zope-instance/issues/5#issuecomment-1313436996

File: cookiecutter.json [hunk 1]
```diff
@@ -9,9 +9,9 @@
     "wsgi_max_request_body_size": "1073741824",
     "wsgi_clear_untrusted_proxy_headers": false,
 
+    "environment_CHAMELEON_CACHE": "{{ cookiecutter.location_clienthome }}/cache",
```
Comment (jensens) [inline cookiecutter.json:12]: I like the idea of not using a dict here, but a namespace like thing.
IMO then it should be done this way for all with prefix `environment_` in the same way

**Review by jensens — APPROVED** [review]
LGTM, but would be nice to have `environment` handled consistently.

File: cookiecutter.json (hunk 1 above)
Comment (me-kell) [reply cookiecutter.json:12]: I agree

File: {{ cookiecutter.target }}/etc/zope.conf [hunk 2]
```diff
@@ -11,14 +11,10 @@ verbose-security {{ "on" if cookiecutter.verbose_security =="True" else "off" }}
...
 default-zpublisher-encoding utf-8
 
-{%- if cookiecutter.environment %}
 <environment>
-{%- for key, value in cookiecutter.environment|dictsort %}
-    {{ key }} {{ value }}
-{%- endfor %}
-
+    CHAMELEON_CACHE {{ cookiecutter.environment_CHAMELEON_CACHE | abspath }}
+    zope_i18n_compile_mo_files {{ cookiecutter.environment_zope_i18n_compile_mo_files }}
```
Comment (jensens) [inline {{ cookiecutter.target }}/etc/zope.conf:16]: Here a feature is lost, adding arbitrary environment strings. So we should loop over all strings starting with `environment_`, right/

**Comment by me-kell** [comment]
Are there any open questions or to dos? Can this PR be merged?

**Comment by jensens** [comment]
> Are there any open questions or to dos? Can this PR be merged?

As mentioned in the review, there is a feature lost and we need a solution here before merge.

**Comment by me-kell** [comment]
Actually I have no idea how to solve both requirements (on the one side to handle path variables and on the other side to use a generic loop for string based environment variables) without marking and knowing the type of the variable.

One workaround could be to declare the path variables (in this case "CHAMELEON_CACHE") separated from the environment variables. Thus looping through the environment and treating them as strings and declaring CHAMELEON_CACHE separately (i.e. outside of the "environment" dictionary) and treating it as path.

IMHO we pay a high price (wrong paths) for having an elegant generic handling of (at the moment) only two (actually only one if we treat CHAMELEON_CACHE separately) variables.

Any idea?

**Comment by jensens** [comment]
I integrated your branch with some modifications around environment handling into toward-2.0 branch.

**Pull request closed without merging**